File: apps/mobile/src/api/activities.ts
import { api } from "./client";
import type { Activity } from "../types";

export const activitiesApi = {
  recent: (token: string) => api.get<Activity[]>("/activities/recent", token),
  create: (
    token: string,
    payload: {
      title: string;
      category: string;
      duration_minutes?: number;
      occurred_at: string;
    }
  ) => api.post<Activity>("/activities", payload, token),
};

File: apps/backend/app/api/activities.py
import uuid

from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

from app.deps import get_current_user_id, get_db
from app.models.activity import Activity, ActivitySource
from app.schemas.activity import ActivityCreate, ActivityOut
from app.services import activity_service

router = APIRouter()


@router.get("/recent", response_model=list[ActivityOut])
def list_recent_activities(
    user_id: uuid.UUID = Depends(get_current_user_id),
    db: Session = Depends(get_db),
) -> list[Activity]:
    return activity_service.get_recent_activities(db, user_id)


@router.post("", response_model=ActivityOut, status_code=status.HTTP_201_CREATED)
def create_activity(
    payload: ActivityCreate,
    user_id: uuid.UUID = Depends(get_current_user_id),
    db: Session = Depends(get_db),
) -> Activity:
    return activity_service.create_activity(
        db,
        user_id,
        title=payload.title,
        category=payload.category,
        duration_minutes=payload.duration_minutes,
        occurred_at=payload.occurred_at,
        source=ActivitySource.manual,
    )
